lib/entities/entity.ts

```diff
@@ -47,7 +47,7 @@ interface EntityParams {
 
 class Entity {
   public id: string;
-  public name: string | Record<string, string>;
+  public name: string | Record<string, string> | undefined | null;
```

Comment:
The entity `name` field may not be changed!
- This must be a string or a Object<string, string> (Record in TS?)
- The `name` field is mandatory for an Entity and may never be null or undefined! Otherwise this will lead to runtime issues.
- Eventhough we assign the return value of the `toLanguageObject` function to this field , which can theoretically return null, it will never return null:
  - the constructor of the Entity base class restricts the input type
  - the deriveded Entity classes also restrict the name input parameter

Note: for extra safety and to verify this contract, an assert statement would make sense, that `name` is available after the assignement in the constructor.

Reply:
Entity definition: https://github.com/unfoldedcircle/core-api/blob/main/integration-api/UCR-integration-asyncapi.yaml#L1668-L1712